oteapi_asmod/strategies/function.py

```diff
@@ -0,0 +1,121 @@
+    # Also, only one model is possible so this should maybe not be
+    # an option.
+    datamodel: Optional[Union[HttpUrl, pathlib.Path]] = Field(
+        "http://onto-ns.com/meta/0.1/Molecule",
+        description=("The dlite datamodel for the structure."),
+    )
+    label: str = Field(
+        ...,
+        description=("Label of the molecule in the dlite collection."),
+    )
+    datacacheKey: str = Field(
+        ...,
+        description=("Key to ase.Atoms obejct in datacache"),
+    )
+    # functionType: str = Field(
+    #        "asedlite/atoms", # shouldn't this be recognized automatically?
+    #        description=("Type of function"),
+    #        )
+
+
+@dataclass
+class ASEDliteFunctionStrategy:
+    """Mapping Strategy."""
+
+    function_config: "FunctionConfig"
+
+    def initialize(
+        self, session: "Optional[Dict[str, Any]]" = None
+    ) -> "Dict[str, Any]":
+        """Initialize strategy.
+
+        This method will be called through the `/initialize` endpoint of the OTE-API
+        Services.
+
+        Parameters:
+            session: A session-specific dictionary context.
+
+        Returns:
+            Dictionary of key/value-pairs to be stored in the sessions-specific
+            dictionary context.
+
+        """
+        return {}
+
+    def get(self, session: "Dict" = None) -> SessionUpdate:
+        """Execute the strategy.
+
+        This method will be called through the strategy-specific endpoint of the
+        OTE-API Services.
+
+        Parameters:
+
+            session: A session-specific dictionary context.
+
+        Returns:
+            Populates session collection with molecule.
+            Returns SessionUpdate()
+
+        """
+        model = self.function_config
+
+        # In ResourceConfig there is a DataCache called configuration
+        # that is used to get out key/value pairs. Is this something that
+        # should be in all Configs?
+        # model = ASEDliteConfig(
+        #    **self.function_config.configuration, extra=Extra.ignore,
+        # )
+
+        # There should be a local folder with entitites at least until onto-ns is up
+        # dlite.storage_path.append(str(pathlib.Path(__file__).parent.resolve()))
+
+        # Create dlite instance of metadata
+        moleculemodel = Instance.create_from_url(
+            "json://" + str(model.datamodel),
+        )  # DLite Metadata
+
+        # Get ase.Atoms obejct from cache
+        cache = DataCache()
+        atoms = cache.get(model.datacacheKey)
+
+        # Creat dlite instance from metadata and populate
+        inst = moleculemodel(dims=[len(atoms), 3], id=model.label)  # DLite instance
+        inst.symbols = atoms.get_chemical_symbols()
+        inst.masses = atoms.get_masses()
+        inst.positions = atoms.positions
+        inst.groundstate_energy = 0.0
+
+        # Get collection from session and place  molecule in it
+        if isinstance(session, dict):
```

Comment:
I guess that `session` may also be an `AttrDict`. Safer to simply check that it is not None (alternatively one could write `isinstance(session, Mapping)` but that currently doesn't work).
```suggestion
        if session is not None:
```